motionplan/collision.go

```diff
@@ -103,15 +103,22 @@ func (oce *ObjectCollisionEntities) indexFromName(name string) (int, error) {
 	return -1, fmt.Errorf("collision entity %q not found", name)
 }
 
-func (oce *ObjectCollisionEntities) checkCollision(key, test *collisionEntity) (float64, error) {
-	distance, err := key.geometry.DistanceFrom(test.geometry)
-	return -distance, err // multiply distance by -1 so that weights of edges are positive
+func (oce *ObjectCollisionEntities) checkCollision(key, test *collisionEntity, depth bool) (float64, error) {
+	if depth {
+		distance, err := key.geometry.DistanceFrom(test.geometry)
+		return -distance, err // multiply distance by -1 so that weights of edges are positive
+	}
+	col, err := key.geometry.CollidesWith(test.geometry)
+	if col {
+		return 1, err
+	}
+	return -1, err
 }
 
 func (oce *ObjectCollisionEntities) reportCollisions(distances []float64) []int {
 	var collisionIndices []int
 	for i := range distances {
-		if distances[i] >= 0 {
+		if distances[i] >= -spatial.CollisionBuffer {
```

Comment:
[major] If we're going to rework the collision system to support a buffer this isn't the way to do it.  The issue here is that the distance between boxes is calculated as an upper limit of the true distance so doing it this way will result in a lot of false positives.  It would be better to do this upon instantiation of the collision system and change the Geometries/GeometryCreators to have the inflated dimensions.  I think this is probably an improvement we can push off to another PR though because this is a sizable feature to add to an already sizable PR

Reply:
Discussed offline; conclusion was we would reduce the CollisonBuffer to 1e-8